### PR TITLE
🔧 CLM-64 A Link That Navigates User From Chats To Profile

### PR DESCRIPTION
**Pull request by JAPHETHNYARANGA**
# Description

his pull request introduces a user-friendly feature that allows users to conveniently navigate to a learner's profile page directly from within the chat interface. Instead of having to access the Students tab and navigate through multiple pages, users can now click on the learner's name r at the top of the chat, and it will open the corresponding learner's profile page.

This enhancement provides users with quick access to detailed learner information, improving their overall experience and efficiency.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Access the chat interface.
Identify a learner's name  displayed at the top.
Click on the learner's name .

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Still some work. Make changes as discussed

File: libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts [hunk 1]
```diff
@@ -78,6 +85,20 @@ export class ChatDetailHeaderComponent implements OnChanges, OnDestroy {
...
       }
     }
+
+    this._enrolledLearners
+      .getLearnerId$(PlatformType.WhatsApp, this.chat.id)
+      .subscribe(
+        (learners: EnrolledEndUser[]) => {
+            const learner = learners[0];
+            // Check if learner.id is defined before passing it a string since learner.id is nullable
+            if (learner.id !== undefined) {
```
Comment (ReaganCn) [inline libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts:95]: Please fix all instances of this

File: libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts [hunk 2]
```diff
@@ -78,6 +85,20 @@ export class ChatDetailHeaderComponent implements OnChanges, OnDestroy {
...
+              this.extractedLearnerId = learner.id.toString()
```
Comment (ReaganCn) [inline libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts:96]: And this

**Review by ReaganCn — CHANGES_REQUESTED** [review]

**Review by ReaganCn — APPROVED** [review]
Good progress

File: libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts [hunk 3]
```diff
@@ -258,6 +278,19 @@ export class ChatDetailHeaderComponent implements OnChanges, OnDestroy {
...
 
+  navigateToLearner(){
+    this._router.navigate([`/learners/${this.extractedLearnerId }`]);
+  }
+
+  updateChatNameColor() {
+    // Update chatNameColor based on avatarBgColor or any other logic you need
+    this.chatNameColor = this.avatarBgColor;
+  }
+  resetChatNameColor() {
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts:289]: one of these variables is redundant. use one

File: libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts [hunk 4]
```diff
@@ -44,6 +47,7 @@ export class ChatDetailHeaderComponent implements OnChanges, OnDestroy {
   @Input() currentPosition: EndUserPosition;
 
   private _sbs = new SubSink();
+  extractedLearnerId: string;// This variable will be used to store the ID of a learner extracted from enrolled learners.
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/conversations/chats/src/lib/components/chat-detail-header/chat-detail-header.component.ts:50]: This name is self-descriptive. No need for comment

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Comment by LemmyMwaura** [comment]
Closing this for now

**Pull request closed without merging**